src/pyk/kcfg/explore.py

```diff
@@ -377,14 +377,24 @@ def extend(
         if self._check_abstract(node, kcfg):
             return
 
-        if not kcfg.splits(target_id=node.id):
-            branches = self.kcfg_semantics.extract_branches(node.cterm)
-            if branches:
-                kcfg.split_on_constraints(node.id, branches)
-                _LOGGER.info(
-                    f'Found {len(branches)} branches using heuristic for node {node.id}: {shorten_hashes(node.id)}: {[self.kprint.pretty_print(bc) for bc in branches]}'
-                )
-                return
+        branches = self.kcfg_semantics.extract_branches(node.cterm)
+
+        branch_conditions = []
+        branch_cterms = []
+
+        for constraint in branches:
+            cterm = node.cterm.add_constraint(constraint)
+            cterm, _ = self.cterm_simplify(cterm)
+            if not cterm.is_bottom:
+                branch_conditions.append(constraint)
+                branch_cterms.append(cterm)
+
+        if len(branch_cterms) > 1 and branches:
+            kcfg.branch(node.id, zip(branch_cterms, branch_conditions, strict=True))
+            _LOGGER.info(
+                f'Found {len(branches)} branches using heuristic for node {node.id}: {shorten_hashes(node.id)}: {[self.kprint.pretty_print(bc) for bc in branches]}'
+            )
+            return
 
```

Comment:
```suggestion
        _branches = self.kcfg_semantics.extract_branches(node.cterm)
        branches = []
        for constraint in _branches:
            cterm = node.cterm.add_constraint(constraint)
            cterm, _ = self.cterm_simplify(cterm)
            if not cterm.is_bottom:
                branches.append(constraint)
        if len(branches) > 1:
            kcfg.split_on_constraints(node.id, branches)
            _LOGGER.info(
                f'Found {len(branches)} branches using heuristic for node {node.id}: {shorten_hashes(node.id)}: {[self.kprint.pretty_print(bc) for bc in branches]}'
            )
            return

```

Does this work? Seems a bit simpler and doesn't require creating a zip list or modifynig `KCFG`.

Reply:
Then we can get rid of the modification down lower too.

Reply:
That would work, that's basically what I did at first. It's just that the new nodes wouldn't have the simplification applied. Maybe we don't care if it immediately gets applied, just that the nodes don't get created if they *would* simplify to bottom?

Reply:
I've changed it to the simpler way for now.